File: optifi-sdk/types/position.ts
import * as anchor from "@project-serum/anchor";
import { PublicKey } from "@solana/web3.js";

export default interface Position {
    instruments: PublicKey,
    longTokenVault: PublicKey,
    shortTokenVault: PublicKey,
    latestPosition: anchor.BN,
    usdcBalance: anchor.BN
}

File: Optifi/programs/Optifi/src/state/position.rs
use anchor_lang::prelude::*;
use anchor_lang::{AnchorDeserialize, AnchorSerialize};
use solana_program::pubkey::Pubkey;

#[derive(Default, Clone, Copy, AnchorDeserialize, AnchorSerialize)]
pub struct Position {
    /// a list of instruments Pubkey corresponded to positions
    pub instruments: Pubkey,
    /// base token account address
    pub long_token_vault: Pubkey,
    /// base token account address
    pub short_token_vault: Pubkey,
    /// latest position updated by update_amm_positions
    pub latest_position: u64,
    /// the usdc remains in the vault
    pub usdc_balance: u64,
}
